src/coord/src/catalog.rs

```diff
@@ -629,10 +629,18 @@ impl Catalog {
     }
 
     /// Returns an iterator over the name of each database in the catalog.
-    pub fn databases(&self) -> impl Iterator<Item = &str> {
+    pub fn list_databases(&self) -> impl Iterator<Item = &str> {
         self.by_name.keys().map(String::as_str)
     }
 
+    /// Returns a (name, id) pair for each database in the catalog.
+    pub fn databases(&self) -> Vec<(String, i64)> {
```

Comment:
Can you make this return `&str` just so that a future caller doesn't need to take the allocation if they don't want it?